src/synology_dsm/api/photos/__init__.py

```diff
@@ -75,6 +76,44 @@ async def get_items_from_album(
             )
         return items
 
+    async def get_items_from_shared_space(
+        self, offset: int = 0, limit: int = 100
+    ) -> list[SynoPhotosItem] | None:
+        """Get a list of all items from the shared space."""
+        items: list[SynoPhotosItem] = []
+        raw_data = await self._dsm.get(
+            self.BROWSE_ITEM_FOTOTEAM_API_KEY,
+            "list",
+            {
+                "offset": offset,
+                "limit": limit,
+                "additional": '["thumbnail"]',
+            },
+        )
+        if not isinstance(raw_data, dict) or (data := raw_data.get("data")) is None:
+            return None
+
+        for item in data["list"]:
+            if item["additional"]["thumbnail"]["xl"] == "ready":
+                size = "xl"
+            elif item["additional"]["thumbnail"]["m"] == "ready":
+                size = "m"
+            else:
+                size = "sm"
+
+            items.append(
+                SynoPhotosItem(
+                    item["id"],
+                    item["type"],
+                    item["filename"],
+                    item["filesize"],
+                    item["additional"]["thumbnail"]["cache_key"],
+                    size,
+                    item["owner_user_id"] == 0,
+                )
+            )
+        return items
```

Comment:
i think we should move this part in a separate function and reuse it in `get_items_from_album`, `get_items_from_shared_space` and `get_items_from_search`